src/app/main/service/route-guards/auth-page-guard.service.ts

```diff
@@ -11,41 +11,42 @@ import { MatDialog } from '@angular/material/dialog';
 })
 export class AuthPageGuardService implements CanActivate {
   private isLoggedIn = false;
-  private ubsRegValue: boolean;
 
   constructor(private localStorageService: LocalStorageService, private router: Router, public dialog: MatDialog) {
     this.localStorageService.userIdBehaviourSubject.subscribe((userId) => (this.isLoggedIn = userId !== null && !isNaN(userId)));
-    this.localStorageService.ubsRegBehaviourSubject.subscribe((value) => (this.ubsRegValue = value));
   }
 
   canActivate(
     route: ActivatedRouteSnapshot,
     state: RouterStateSnapshot
   ): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
     if (!this.isLoggedIn) {
-      this.openSingInWindow('sign-in');
-      return of(false);
+      this.openSingInWindow('sign-in', state);
+      if (state.url.includes('ubs')) {
```

Comment:
const path = state.url.includes('ubs') ? '/ubs' :  '/greenCity'
return this.router.parseUrl(path);

Reply:
fixed